src/DevBetterWeb.Infrastructure/Services/DailyCheckPingService.cs

```diff
@@ -64,11 +68,49 @@ public async Task SendPingIfNeeded(AppendOnlyStringList messages)
       }
     }
 
+    public async Task PingAdminsAboutAlmostAlumsIfNeeded(AppendOnlyStringList messages)
+    {
+      var members = await _memberRepository.ListAsync();
+      var membersToPingAdminsAbout = new List<Member>();
+
+      var daysBeforeGraduationToPing = 10;
+
+      foreach (var member in members)
+      {
+        if (member.TotalSubscribedDays() == DAYS_IN_TWO_YEARS - daysBeforeGraduationToPing)
+        {
+          membersToPingAdminsAbout.Add(member);
+        }
+      }
+
+      if (membersToPingAdminsAbout.Any())
+      {
+
+        string listOfMembersToPingAdminsAbout = "";
+
+        foreach (var member in membersToPingAdminsAbout)
+        {
+          listOfMembersToPingAdminsAbout = $"{member.UserFullName()}\n";
+        }
+
+        messages.Append($"Admins need to be reminded not to renew the subscriptions of the following members about to graduate: {listOfMembersToPingAdminsAbout}");
+
+        var usersInAdminRole = await _userManager.GetUsersInRoleAsync(AuthConstants.Roles.ADMINISTRATORS);
+
+        foreach(var user in usersInAdminRole)
+        {
+          await _emailService.SendEmailAsync(user.Email, "Upcoming DevBetter Graduation", $"Ensure that the following member(s)'s subscriptions are not renewed, as they will graduate to alumni status in {daysBeforeGraduationToPing} days:\n {listOfMembersToPingAdminsAbout}");
+        }
+
+        messages.Append($"Admins have been reminded not to renew the subscriptions of the following members about to graduate: {listOfMembersToPingAdminsAbout}");
+      }
+    }
+
```

Comment:
This class is getting long and already violates SRP. If we end up adding another method to it, I would want to see it refactored so that at least the body of each of these methods was implemented by another service (or handler) and this service simply delegated work out to other services/handlers as needed (and would thus become a lot shorter).

Reply:
ok